### PR TITLE
New cool stuff

### PR DESCRIPTION
**Pull request by dylan-muszel**
All tests passed:
image

### CODE REVIEW

File: core/src/main/java/ar/com/wolox/wolmo/core/extensions/GenericsExtensions.kt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package ar.com.wolox.wolmo.core.extensions
+
+fun <T> T?.orElse(other: T) = this ?: other
+
+val <T> T?.unit
```
Comment (lmiotti) [inline core/src/main/java/ar/com/wolox/wolmo/core/extensions/GenericsExtensions.kt]: Why I would like to receive an `Unit` type? I don't understand the aim of this extension

File: core/src/main/java/ar/com/wolox/wolmo/core/util/WolmoFileProvider.kt [hunk 2]
```diff
@@ -38,6 +42,61 @@ import javax.inject.Inject
...
+    private val appName: String by lazy {
+        val applicationInfo = context.applicationInfo
+        val stringId = applicationInfo.labelRes
+        if (stringId != 0) context.getString(stringId) else applicationInfo.nonLocalizedLabel?.toString() ?: ""
+    }
+
+    private var cacheFolder: String = getTmpDirectory(context).absolutePath
+
+    private fun getTmpDirectory(context: Context) = File(context.cacheDir, appName).apply {
+        if (!exists()) {
```
Comment (lmiotti) [inline core/src/main/java/ar/com/wolox/wolmo/core/util/WolmoFileProvider.kt]: Maybe we can write this condition in one line

File: core/src/main/java/ar/com/wolox/wolmo/core/util/WolmoFileProvider.kt [hunk 3]
```diff
@@ -38,6 +42,61 @@ import javax.inject.Inject
...
+    private fun getEnvironmentFilename(folder: String, name: String, extension: String, mime: String): String? {
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            val resolver = context.contentResolver
+            val contentValues = ContentValues().apply {
+                put(MediaStore.MediaColumns.DISPLAY_NAME, FILENAME_FORMAT.format(name, System.nanoTime()))
+                put(MediaStore.MediaColumns.MIME_TYPE, mime)
+                put(MediaStore.MediaColumns.RELATIVE_PATH, "$folder/$appName")
+            }
+
+            resolver.insert(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, contentValues)?.let {
```
Comment (lmiotti) [inline core/src/main/java/ar/com/wolox/wolmo/core/util/WolmoFileProvider.kt]: Between `resolver` creation and this line, it's not used. we can do
`context.contentResolver.insert(....)` instead. What do you think?

File: core/src/main/java/ar/com/wolox/wolmo/core/util/WolmoFileProvider.kt [hunk 4]
```diff
@@ -38,6 +42,61 @@ import javax.inject.Inject
...
+                getRealPathFromUri(it)
+            }
+        } else {
+            val file = File(Environment.getExternalStoragePublicDirectory(folder), appName)
```
Comment (lmiotti) [inline core/src/main/java/ar/com/wolox/wolmo/core/util/WolmoFileProvider.kt]: We can use some `Kotlin` standard function?

File: core/src/main/java/ar/com/wolox/wolmo/core/extensions/GenericsExtensions.kt (hunk 1 above)
Comment (dylan-muszel) [reply core/src/main/java/ar/com/wolox/wolmo/core/extensions/GenericsExtensions.kt]: Oooh, right! I wanted to discuss it before and I forgot. We'll discuss it in private 😄.

File: core/src/test/java/ar/com/wolox/wolmo/core/extensions/ViewExtensionsTest.kt [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        textView.setTextOrGone("")
+
+        assertThat(textView.isVisible, `is`(false))
+        assertThat(textView.text.toString(), `is`(""))
+    }
+
+    @Test
+    fun `given a visible TextView when set text or gone a non empty text then the TextView shows text`() {
+        val textView = TextView(ApplicationProvider.getApplicationContext())
+        assertThat(textView.isVisible,       `is`(true))
```
Comment (igsosa92) [inline core/src/test/java/ar/com/wolox/wolmo/core/extensions/ViewExtensionsTest.kt]: why is there so much blank spaces here? xD I just left this as a commet, is all approved :p

File: core/build.gradle [hunk 6]
```diff
@@ -50,6 +52,8 @@ buildscript {
     ext.mockito_inline_version = '2.13.0'
     ext.assertj_version = '3.9.0'
     ext.android_ktx_version = '1.1.0'
+    ext.coroutines_version = '1.3.3'
+    ext.material_version = '1.0.0'
```
[the remark this answers is not in the document]
Comment (dylan-muszel) [reply core/build.gradle]: I removed it because it was unused. 😄

**Pull request merged**